### PR TITLE
Update sqlb_de.ts

### PR DESCRIPTION
**Pull request by Mr-Update**
- Spelling and grammar corrected
- Spelling standardised
- Translation improved
- Two superfluous shortcuts (Ctrl+D and Ctrl I) removed
- One type="unfinished" removed

### CODE REVIEW

File: src/translations/sqlb_de.ts [hunk 1]
```diff
@@ -365,12 +365,12 @@ Attribution Share Alike 4.0 license.&lt;br/&gt;See &lt;/span&gt;&lt;a href=&quot
     <message>
         <location filename="../CipherDialog.ui" line="110"/>
         <source>SQLCipher &amp;3 defaults</source>
-        <translation>SQLCipher &amp;3 Standardwerte</translation>
+        <translation>SQLCipher &amp;3-Standardwerte</translation>
```
Comment (FriedrichFroebel) [inline src/translations/sqlb_de.ts:368]: I would argue against this unless we add a dash before the number as well.

File: src/translations/sqlb_de.ts [hunk 2]
```diff
@@ -365,12 +365,12 @@ Attribution Share Alike 4.0 license.&lt;br/&gt;See &lt;/span&gt;&lt;a href=&quot
...
     </message>
     <message>
         <location filename="../CipherDialog.ui" line="117"/>
         <source>SQLCipher &amp;4 defaults</source>
-        <translation>SQLCipher &amp;4 Standardwerte</translation>
+        <translation>SQLCipher &amp;4-Standardwerte</translation>
```
Comment (FriedrichFroebel) [inline src/translations/sqlb_de.ts:373]: I would argue against this unless we add a dash before the number as well.

File: src/translations/sqlb_de.ts [hunk 3]
```diff
@@ -2393,7 +2393,7 @@ x~y	Bereich: Werte zwischen x und y
     <message>
         <location filename="../FindReplaceDialog.ui" line="96"/>
         <source>When set, the search goes backwards from cursor position, otherwise it goes forward</source>
-        <translation>Falls gesetzt, erfolgt die Suche rückwärts von der Cursorposition, andernfalls erfolgt sie vorwärts</translation>
+        <translation>Wenn gesetzt, erfolgt die Suche rückwärts von der Cursorposition, anderenfalls erfolgt sie vorwärts</translation>
```
Comment (FriedrichFroebel) [inline src/translations/sqlb_de.ts:2396]: Is the second change really necessary. I personally have never used/seen the new word before, although being a native speaker.

File: src/translations/sqlb_de.ts [hunk 4]
```diff
@@ -3183,7 +3183,7 @@ Sie können SQL-Anweisungen aus einer Objektzeile fassen und in anderen Anwendun
     <message>
         <location filename="../MainWindow.ui" line="2002"/>
         <source>Extra DB Toolbar</source>
-        <translation>Extra-DB-Werkzeugleiste</translation>
+        <translation>Extra DB-Werkzeugleiste</translation>
```
Comment (FriedrichFroebel) [inline src/translations/sqlb_de.ts:3186]: Is there any reason for this change?

File: src/translations/sqlb_de.ts [hunk 5]
```diff
@@ -3764,7 +3764,7 @@ Sie können SQL-Anweisungen aus der Schemaspalte nehmen und in den SQL-Editor od
     <message>
         <location filename="../MainWindow.ui" line="1143"/>
         <source>Extra DB toolbar</source>
-        <translation>Extra-DB-Werkzeugleiste</translation>
+        <translation>Extra DB-Werkzeugleiste</translation>
```
Comment (FriedrichFroebel) [inline src/translations/sqlb_de.ts:3767]: Is there any reason for this change (see above)?

File: src/translations/sqlb_de.ts [hunk 6]
```diff
@@ -3896,7 +3896,7 @@ Sie können SQL-Anweisungen aus der Schemaspalte nehmen und in den SQL-Editor od
     <message>
         <location filename="../MainWindow.cpp" line="400"/>
         <source>Database encoding</source>
-        <translation>Datenbank-Kodierung</translation>
+        <translation>Datenbank-Codierung</translation>
```
Comment (FriedrichFroebel) [inline src/translations/sqlb_de.ts:3899]: Is there any reason for this change?

File: src/translations/sqlb_de.ts [hunk 7]
```diff
@@ -3984,12 +3984,12 @@ Reason: %1</source>
...
         <location filename="../MainWindow.cpp" line="2456"/>
         <source>A new DB Browser for SQLite version is available (%1.%2.%3).&lt;br/&gt;&lt;br/&gt;Please download at &lt;a href=&apos;%4&apos;&gt;%4&lt;/a&gt;.</source>
-        <translation>Eine neue Version des DB Browsers für SQLite ist verfügbar (%1.%2.%3).&lt;br/&gt;&lt;br/&gt;Bitte laden Sie diese von &lt;a href=&apos;%4&apos;&gt;%4&lt;/a&gt; herunter.</translation>
+        <translation>Eine neue Version des DB-Browsers für SQLite ist verfügbar (%1.%2.%3).&lt;br/&gt;&lt;br/&gt;Bitte laden Sie diese von &lt;a href=&apos;%4&apos;&gt;%4&lt;/a&gt; herunter.</translation>
     </message>
     <message>
         <location filename="../MainWindow.cpp" line="2686"/>
         <source>DB Browser for SQLite project file (*.sqbpro)</source>
-        <translation>DB Browser für SQLite Projektdatei (*.sqbpro)</translation>
+        <translation>DB-Browser für SQLite-Projektdatei (*.sqbpro)</translation>
```
Comment (FriedrichFroebel) [inline src/translations/sqlb_de.ts:3992]: This does not really look better with partially added dashes IMHO.

File: src/translations/sqlb_de.ts [hunk 8]
```diff
@@ -4895,7 +4895,7 @@ Warnung: es wurden aufgrund der partiellen Abrufmechanismus noch nicht alle Date
     <message>
         <location filename="../PreferencesDialog.ui" line="632"/>
         <source>Database &amp;encoding</source>
-        <translation>Datenbank-&amp;Kodierung</translation>
+        <translation>Datenbank-&amp;Codierung</translation>
```
Comment (FriedrichFroebel) [inline src/translations/sqlb_de.ts:4898]: See above.

File: src/translations/sqlb_de.ts [hunk 9]
```diff
@@ -5805,7 +5805,7 @@ Bitte erneut prüfen.</translation>
     <message>
         <location filename="../FileDialog.h" line="16"/>
         <source>DB Browser for SQLite Project Files (*.sqbpro)</source>
-        <translation>DB Browser für SQLite Projektdateien (*.sqbpro)</translation>
+        <translation>DB-Browser für SQLite-Projektdateien (*.sqbpro)</translation>
```
Comment (FriedrichFroebel) [inline src/translations/sqlb_de.ts:5808]: See above.

File: src/translations/sqlb_de.ts [hunk 10]
```diff
@@ -6716,7 +6716,7 @@ Die Verwendung dieser Funktion muss in den Einstellungen authorisiert werden.</t
     <message>
         <location filename="../SqlUiLexer.cpp" line="109"/>
         <source>(X) The soundex(X) function returns a string that is the soundex encoding of the string X.</source>
-        <translation>(X) Die soundex(X)-Funktion gibt eine Zeichenkette zurück, die aus der Soundex-Kodierung von Zeichenkette X besteht.</translation>
+        <translation>(X) Die soundex(X)-Funktion gibt eine Zeichenkette zurück, die aus der Soundex-Codierung von Zeichenkette X besteht.</translation>
```
Comment (FriedrichFroebel) [inline src/translations/sqlb_de.ts:6719]: See above.

File: src/translations/sqlb_de.ts [hunk 11]
```diff
@@ -6840,48 +6840,48 @@ Die Verwendung dieser Funktion muss in den Einstellungen authorisiert werden.</t
...
-        <translation>(expr,offset) Falls das Offset-Argument angegeben ist, dann muss dieses ein nicht-negativer Integerwert sein. In diesem Fall ist der Rückgabewert das Ergebnis der Evaluation von expr gegen die Zeile, die innerhalb der Partition offset Zeilen weiter oben liegt. Falls offset 0 ist, wird expr gegen die aktuelle Zeile evaluiert. Falls vor der aktuellen Zeile nicht genügend Zeilen vorhanden sind, wird NULL zurückgegeben.</translation>
+        <translation>(expr,offset) Wenn das Argument offset angegeben ist, dann muss dieses ein nicht negativer Integerwert sein. In diesem Fall ist der Rückgabewert das Ergebnis der Evaluation von expr gegen die Zeile, die innerhalb der Partition offset Zeilen weiter oben liegt. Wenn offset 0 ist, wird expr gegen die aktuelle Zeile evaluiert. Wenn vor der aktuellen Zeile nicht genügend Zeilen vorhanden sind, wird NULL zurückgegeben.</translation>
```
Comment (FriedrichFroebel) [inline src/translations/sqlb_de.ts:6853]: I would probably prefer `nichtnegativer` here.

File: src/translations/sqlb_de.ts [hunk 12]
```diff
@@ -6840,48 +6840,48 @@ Die Verwendung dieser Funktion muss in den Einstellungen authorisiert werden.</t
...
-        <translation>(expr,offset) Falls das Offset-Argument angegeben ist, dann muss dieses ein nicht-negativer Integerwert sein. In diesem Fall ist der Rückgabewert das Ergebnis der Evaluation von expr gegen die Zeile, die innerhalb der Partition offset Zeilen weiter unten liegt. Falls offset 0 ist, wird expr gegen die aktuelle Zeile evaluiert. Falls nach der aktuellen Zeile nicht genügend Zeilen vorhanden sind, wird NULL zurückgegeben.</translation>
+        <translation>(expr,offset) Wenn das Argument offset angegeben ist, dann muss dieses ein nicht negativer Integerwert sein. In diesem Fall ist der Rückgabewert das Ergebnis der Evaluation von expr gegen die Zeile, die innerhalb der Partition offset Zeilen weiter unten liegt. Wenn offset 0 ist, wird expr gegen die aktuelle Zeile evaluiert. Wenn nach der aktuellen Zeile nicht genügend Zeilen vorhanden sind, wird NULL zurückgegeben.</translation>
```
Comment (FriedrichFroebel) [inline src/translations/sqlb_de.ts:6869]: I would probably prefer `nichtnegativer` here.

File: src/translations/sqlb_de.ts [hunk 13]
```diff
@@ -7510,22 +7510,22 @@ Halten Sie %3Umschalt und klicken Sie, um hierher zu springen</translation>
         <location filename="../TableBrowser.ui" line="738"/>
         <location filename="../TableBrowser.cpp" line="1497"/>
         <source>Set encoding</source>
-        <translation>Kodierung setzen</translation>
+        <translation>Codierung setzen</translation>
```
Comment (FriedrichFroebel) [inline src/translations/sqlb_de.ts:7513]: See above.

File: src/translations/sqlb_de.ts [hunk 14]
```diff
@@ -7510,22 +7510,22 @@ Halten Sie %3Umschalt und klicken Sie, um hierher zu springen</translation>
...
     </message>
     <message>
         <location filename="../TableBrowser.ui" line="741"/>
         <source>Change the encoding of the text in the table cells</source>
-        <translation>Kodierung des Textes in den Tabellenzellen ändern</translation>
+        <translation>Codierung des Textes in den Tabellenzellen ändern</translation>
```
Comment (FriedrichFroebel) [inline src/translations/sqlb_de.ts:7518]: See above.

File: src/translations/sqlb_de.ts [hunk 15]
```diff
@@ -7510,22 +7510,22 @@ Halten Sie %3Umschalt und klicken Sie, um hierher zu springen</translation>
...
     </message>
     <message>
         <location filename="../TableBrowser.ui" line="746"/>
         <source>Set encoding for all tables</source>
-        <translation>Kodierung für alle Tabellen setzen</translation>
+        <translation>Codierung für alle Tabellen setzen</translation>
```
Comment (FriedrichFroebel) [inline src/translations/sqlb_de.ts:7523]: See above.

File: src/translations/sqlb_de.ts [hunk 16]
```diff
@@ -7510,22 +7510,22 @@ Halten Sie %3Umschalt und klicken Sie, um hierher zu springen</translation>
...
     </message>
     <message>
         <location filename="../TableBrowser.ui" line="749"/>
         <source>Change the default encoding assumed for all tables in the database</source>
-        <translation>Voreingestellte Kodierung für alle Tabellen in der Datenbank ändern</translation>
+        <translation>Voreingestellte Codierung für alle Tabellen in der Datenbank ändern</translation>
```
Comment (FriedrichFroebel) [inline src/translations/sqlb_de.ts:7528]: See above.

File: src/translations/sqlb_de.ts [hunk 17]
```diff
@@ -7858,24 +7858,24 @@ Halten Sie %3Umschalt und klicken Sie, um hierher zu springen</translation>
     <message>
         <location filename="../TableBrowser.cpp" line="1493"/>
         <source>Please choose a new encoding for all tables.</source>
-        <translation>Bitte wählen Sie eine neue Kodierung für alle Tabellen.</translation>
+        <translation>Bitte wählen Sie eine neue Codierung für alle Tabellen.</translation>
```
Comment (FriedrichFroebel) [inline src/translations/sqlb_de.ts:7861]: See above.

File: src/translations/sqlb_de.ts [hunk 18]
```diff
@@ -7858,24 +7858,24 @@ Halten Sie %3Umschalt und klicken Sie, um hierher zu springen</translation>
...
     </message>
     <message>
         <location filename="../TableBrowser.cpp" line="1495"/>
         <source>Please choose a new encoding for this table.</source>
-        <translation>Bitte wählen Sie eine neue Kodierung für diese Tabelle.</translation>
+        <translation>Bitte wählen Sie eine neue Codierung für diese Tabelle.</translation>
```
Comment (FriedrichFroebel) [inline src/translations/sqlb_de.ts:7866]: See above.

File: src/translations/sqlb_de.ts [hunk 19]
```diff
@@ -7858,24 +7858,24 @@ Halten Sie %3Umschalt und klicken Sie, um hierher zu springen</translation>
...
     </message>
     <message>
         <location filename="../TableBrowser.cpp" line="1498"/>
         <source>%1
 Leave the field empty for using the database encoding.</source>
         <translation>%1
-Lassen Sie das Feld leer, um die Datenbank-Kodierung zu verwenden.</translation>
+Lassen Sie das Feld leer, um die Datenbank-Codierung zu verwenden.</translation>
```
Comment (FriedrichFroebel) [inline src/translations/sqlb_de.ts:7873]: See above.

File: src/translations/sqlb_de.ts [hunk 20]
```diff
@@ -7858,24 +7858,24 @@ Halten Sie %3Umschalt und klicken Sie, um hierher zu springen</translation>
...
     </message>
     <message>
         <location filename="../TableBrowser.cpp" line="1510"/>
         <source>This encoding is either not valid or not supported.</source>
-        <translation>Diese Kodierung ist entweder nicht gültig oder nicht unterstützt.</translation>
+        <translation>Diese Codierung ist entweder nicht gültig oder nicht unterstützt.</translation>
```
Comment (FriedrichFroebel) [inline src/translations/sqlb_de.ts:7878]: See above.

**Comment by Mr-Update** [comment]
@FriedrichFroebel

**DB-Browser für SQLite**
`DB-Browser für SQLite` has already been used 8 times. I have only added the hyphen of the 7 occurrences of `DB Browser für SQLite` to standardize them.
As it has been Germanized (`für` instead of `for`), the Germanization should be complete.
However, program names are actually proper names and are not translated. This opinion is also shared by mgrojo in discussion 3530.
<br>
**Spelling recommended by Duden**
`Codierung` see https://www.duden.de/rechtschreibung/Codierung
`anderenfalls` see https://www.duden.de/rechtschreibung/anderenfalls
`nicht negativ` see https://www.duden.de/rechtschreibung/nicht_deutsch as an example of the recommended separate spelling.
```
If there are two or more spelling variants, the variant recommended by the Duden editors is shown first.
```
(from https://www.duden.de/hilfe/rechtschreibung)
<br>
**Extra DB-Werkzeugleiste**
I understand `Extra` as an adjective in the sense of `Additional DB toolbar`.
<br>
**SQLCipher 3-Standardwerte**
**SQLCipher 4-Standardwerte**
The "German Localization Style Guide" (see https://www.microsoft.com/en-us/language/styleguides) states at the top of page 27:
```
Product names are not hyphenated for legal reasons. However, in order to offset
the actual product name from the word that follows in a compound, the hyphen is
placed right between the product name and the following term. Please note that
this rule applies even if the word that follows is an English term. The
only exception is a product name directly followed by the word “Setup”, because
this Setup is treated like a product name.

Examples:
Skype Manager-Administrator
Skype-Version
```

Further examples can be found under "Microsoft UI String Search" (see https://learn.microsoft.com/en-us/globalization/reference/microsoft-language-resources):
```
Office 365-Anwendung
Microsoft Teams-Protokoll
Windows 10-Updates
```

**Comment by FriedrichFroebel** [comment]
Thanks for the explanations. The only thing I still do not agree with is `nichtnegativ` - I would consider the expressions to be more or less mathematical and thus use the mathematical version here: https://de.wiktionary.org/wiki/nichtnegativ

**Comment by Mr-Update** [comment]
Changed twice to `nichtnegativ`

**Comment by justinclift** [comment]
Oh, just noticed this is still not merged.

Um, @FriedrichFroebel were you happy with this before it got kind of missed?

GitHub's system is saying there are conflicts now, but they might not be too hard to fix.  At a rough guess, that's probably just related to this commit (b4c46caa401f771aeeeb54826479c5c99e2e20f7).

**Comment by lucydodo** [comment]
The conflicts was caused by a PR #3572 we had today, but after reviewing it, it didn't seem to be a big deal, so I just fixed it.

**Comment by justinclift** [comment]
Awesome.  If @FriedrichFroebel is happy with this PR, we should merge it and also add it to the branch for our 3.13 release. :smile:

**Review by FriedrichFroebel — APPROVED** [review]

**Comment by justinclift** [comment]
Thanks @FriedrichFroebel.  I'll merge this shortly. :smile:

**Comment by justinclift** [comment]
@lucydodo I manually rebased and merged the first commit to include the updated https links from the PR yesterday.  Doing that seems to have nuked your merge commit somehow. :facepalm:

**Pull request merged**